### PR TITLE
feat: Prove limit of H_{2,3,4}(z) as z -> i\infty

### PR DESCRIPTION
**Pull request by grhkm21**
In this PR, I prove that $\lim_{z \to i\infty} (H_2(z), H_3(z), H_4(z)) = (0, 1, 1)$.

Actually I think I can replace the entire proof of #39 by this.

Depends on: #39

### CODE REVIEW

File: blueprint/src/subsections/modular-forms.tex [hunk 1]
```diff
@@ -293,13 +293,23 @@ \subsection{Modular forms and examples}
...
-  \todo{Fill in this proof, in particular the analytic properties of $H_i$.}
+  We have to show that each $H_i$ satisfies property (2) of \cref{def:holomorphic-modular-form}. It suffices to show that for any $\gamma \in \Gamma_1$, $\|H_2|_2\gamma(z)\|$ is bounded as $z \in \mathbb{H} \to i\infty$. Firstly, by \cref{lemma:theta-modular}, \cref{lemma:Gamma-2-generators} and induction on group elements, we notice that $\{\pm H_2, \pm H_3, \pm H_4\}$ is closed under action by $\Gamma_1$. Hence, it suffices to prove that $H_2$, $H_3$ and $H_4$ are bounded at $i\infty$. We proceed by direct algebraic manipulation:
 
-  We have to show that each $H_i$ satisfies property (2) of \cref{def:holomorphic-modular-form}
+\begin{align}
+  \|H_2(z)\|
+  &= \left\|\sum_{n \in \Z} \exp\left(\pi i \left(n + \frac{1}{2}\right)^2 z\right)\right\|^4
+  \leq \left(\sum_{n \in \Z} \left\|\exp\left(\pi i \left(n + \frac{1}{2}\right)^2 z\right)\right)\right\|^4 \\
+  &= \left(\sum_{n \in \Z} \left\|\exp\left(-\pi \left(n + \frac{1}{2}\right)^2 \Im(z)\right)\right)\right\|^4
+  \leq \left(\sum_{n \in \Z} \left\|\exp\left(-\pi \left(n + \frac{1}{2}\right)^2 A\right)\right)\right\|^4
```
Comment (seewoo5) [inline blueprint/src/subsections/modular-forms.tex:307]: What is $A$ here? maybe $z = iA$?

Comment (grhkm21) [reply blueprint/src/subsections/modular-forms.tex:307]: It's the bound for $\Im(z)$ IIRC. (Sorry I copied thisf rom on paper)

File: SpherePacking/ModularForms/AtImInfty.lean [hunk 2]
```diff
@@ -0,0 +1,109 @@
+import Mathlib.Analysis.Normed.Group.Tannery
+import Mathlib.NumberTheory.ModularForms.JacobiTheta.Bounds
+
+import SpherePacking.ModularForms.JacobiTheta
+
+/-!
+# Limits at infinity
+
+In this file we prove the limit of Θᵢ(z) as z tends to i∞. This will be useful as we do
+computations with Fourier coefficients, e.g. comparing two q-expansions.
```
Comment (seewoo5) [inline SpherePacking/ModularForms/AtImInfty.lean]: This is also useful for the computation of limit later! ($lim_{t \to 0^+} F(it) / G(it) = 18 / \pi^2$ thing)

File: blueprint/src/subsections/modular-forms.tex (hunk 1 above)
Comment (seewoo5) [reply blueprint/src/subsections/modular-forms.tex:307]: I see, maybe the same $A$ in `Filter.eventually_atImInfty`?

File: SpherePacking/ModularForms/JacobiTheta.lean [hunk 3]
```diff
@@ -248,42 +259,226 @@ noncomputable def H₄_SIF : SlashInvariantForm (Γ 2) 2 where
...
+  intro z hz
+  rw [map_pow]
+  gcongr
+  calc
+    _ = ‖∑' (n : ℤ), cexp (π * I * (n + 1 / 2) ^ 2 * z)‖ := rfl
+    _ ≤ ∑' (n : ℤ), ‖cexp (π * I * (n + 1 / 2) ^ 2 * z)‖ := norm_tsum_le_tsum_norm ?_
+    _ = ∑' (n : ℤ), ‖cexp (π * I * ((n + 1 / 2) ^ 2 * z : ℂ))‖ := by simp only [← mul_assoc]
+    _ = ∑' (n : ℤ), ‖rexp (-π * (((n + 1 / 2) ^ 2 : ℝ) * z : ℂ).im)‖ := by
+      apply tsum_congr fun b ↦ ?_
+      have (z : ℂ) : ‖cexp z‖ = ‖cexp z.re‖ := by
```
Comment (seewoo5) [inline SpherePacking/ModularForms/JacobiTheta.lean:323]: I think this is useful lemma, along with `‖cexp (I * z)‖ = rexp (z.im)` is good to have as a separate lemma (you can see #37 for the possible applications of this)

File: blueprint/src/subsections/modular-forms.tex (hunk 1 above)
Comment (grhkm21) [reply blueprint/src/subsections/modular-forms.tex:307]: Yes exactly

File: SpherePacking/ModularForms/JacobiTheta.lean (hunk 3 above)
Comment (grhkm21) [reply SpherePacking/ModularForms/JacobiTheta.lean:323]: This is `Complex.norm_exp_mul_I`

**Pull request merged**